plugin/arcconf/arcconf.py

```diff
@@ -0,0 +1,657 @@
+                                      _fw_version=fw_ver,
+                                      _mode=mode,
+                                      _read_cache_pct=read_cache_pct))
+
+        return rc_lsm_syss
+
+    @staticmethod
+    def _arcconf_array_to_lsm_pool(arcconf_array, ld_name, sys_id, ctrl_num):
+        pool_id = '%s:%s' % (ctrl_num, ld_name)
+        name = 'Array' + str(ld_name)
+        elem_type = Pool.ELEMENT_TYPE_VOLUME | Pool.ELEMENT_TYPE_VOLUME_FULL
+        unsupported_actions = 0
+        free_space = 0
+        total_space = 0
+        # TODO Need to have a logic for finding free space and total space.
+
+        status = Pool.STATUS_OK
+        status_info = ''
+        plugin_data = "%s:%s" % (ctrl_num, ld_name)
+
+        return Pool(
+            pool_id, name, elem_type, unsupported_actions,
+            total_space, free_space, status, status_info,
+            sys_id, plugin_data)
+
+    @_handle_errors
+    def pools(self, search_key=None, search_value=None,
+              flags=Client.FLAG_RSVD):
+        lsm_pools = []
+        total_cntrl = self._get_arcconf_controllers_count()
+        sys_id = ''
+
+        for cntrl in range(total_cntrl):
+            cntrl_num = str(int(cntrl + 1))
+            sys_id = cntrl_num
+
+            disks_info = self._arcconf_exec(["getconfigJSON", str(cntrl+1)])
+            disk_info_json = disks_info.split("\n")[1]
+            decoded_json = json.loads(disk_info_json)
+
+            if 'LogicalDrive' in decoded_json['Controller']:
+                for ld in range(len(decoded_json['Controller']['LogicalDrive'])):
+                    lsm_pools.append(
+                    Arcconf._arcconf_array_to_lsm_pool(
+                    decoded_json['Controller']['LogicalDrive'][ld]['logicalDriveID'],
+                    decoded_json['Controller']['LogicalDrive'][ld]['logicalDriveID'],
+                    sys_id,
+                    cntrl_num))
+        return search_property(lsm_pools, search_key, search_value)
+
+    @staticmethod
+    def _arcconf_ld_to_lsm_vol(arcconf_ld, pool_id, sys_id, ctrl_num, array_num,
+                          arcconf_ld_name):
+        """
+        raises DeviceNotFoundError
+        """
+        ld_num = arcconf_ld['logicalDriveID']
+        vpd83 = str(arcconf_ld['volumeUniqueID']).lower()
+
+        block_size = arcconf_ld['BlockSize']
+        num_of_blocks = int(arcconf_ld['dataSpace']) / int(block_size)
+        vol_name = arcconf_ld_name
+
+        if len(vpd83) > 0:
+            blk_paths = LocalDisk.vpd83_search(vpd83)
+            if len(blk_paths) > 0:
+                blk_path = blk_paths[0]
+                vol_name += ": %s" % " ".join(blk_paths)
+                device = Device.from_device_file(_CONTEXT, blk_path)
+                attributes = device.attributes
+                try:
+                    block_size = attributes.asint("BlockSize")
+                    num_of_blocks = attributes.asint("dataSpace")
```

Comment:
But the `attributes` here is from pyudev and the line here means using sysfs file `BlockSize` and `dataSpace` in sysfs of this block. They are not from your JSON data.